### PR TITLE
I-52 Modify GroupDto & GroupService

### PR DESCRIPTION
**Pull request by LeSitar**
#52

### CODE REVIEW

File: src/main/java/com/school/project/controller/GroupController.java [hunk 1]
```diff
@@ -22,28 +22,37 @@
     private GroupService groupService;
 
     @PostMapping("/group")
-    public Group createGroup(@RequestBody GroupDto groupDto){
-        return groupService.create(convertGroupDtoToGroup(groupDto));
+    public ResponseEntity <Group> createGroup(@RequestBody GroupDto groupDto){
```
Comment (MichelRogov) [inline src/main/java/com/school/project/controller/GroupController.java]: should return GroupDto

File: src/main/java/com/school/project/controller/GroupController.java [hunk 2]
```diff
@@ -22,28 +22,37 @@
...
+        return ResponseEntity.ok()
+                .body(groupService.createUpdate(convertGroupDtoToGroup(groupDto)));
+    }
+
+    @PutMapping("/group/{groupId}/add/{userId}")
+    public void addUserToGroup(@PathVariable Long groupId, @PathVariable Long userId){
+        ResponseEntity.ok().body(convertGroupToGroupDto(groupService.addUser(groupId, userId)));
```
Comment (MichelRogov) [inline src/main/java/com/school/project/controller/GroupController.java]: why do you need body here? it should look like: line 32 - calling service, line 33 - returning Response

File: src/main/java/com/school/project/service/impl/GroupServiceImpl.java [hunk 3]
```diff
@@ -31,8 +39,23 @@ public Group getGroupById(Long id) {
...
+
         return groupRepository.findAll();
     }
 
+    @Override
+    public Group addUser(Long groupId, Long userId) {
+
+        Optional<User> userById = userRepository.findById(userId);
+        if (!userById.isPresent()) {
+            throw new UserNotFoundException("User with id " + userId + " was not found");
```
Comment (MichelRogov) [inline src/main/java/com/school/project/service/impl/GroupServiceImpl.java]: don't repeat your self.
you have already this logic, why not to reuse it?

File: src/main/java/com/school/project/service/impl/GroupServiceImpl.java [hunk 4]
```diff
@@ -31,8 +39,23 @@ public Group getGroupById(Long id) {
...
+        }
+        Group group = getById(groupId);
+        List <User> users = group.getUserList();
+        users.add(userById.get());
+
```
Comment (MichelRogov) [inline src/main/java/com/school/project/service/impl/GroupServiceImpl.java]: save?

File: src/main/java/com/school/project/controller/GroupController.java [hunk 5]
```diff
@@ -22,28 +23,37 @@
...
     @GetMapping("/groups")
     public List<GroupDto> getAllGroups() {
-        return groupService.getAllGroups().stream()
+        return groupService.getAll().stream()
                 .map(s -> convertGroupToGroupDto(s))
                 .collect(Collectors.toList());
     }
 
     private Group convertGroupDtoToGroup(GroupDto groupDto) {
+
```
Comment (MichelRogov) [inline src/main/java/com/school/project/controller/GroupController.java]: get rid of this line here

File: src/main/java/com/school/project/controller/GroupController.java [hunk 6]
```diff
@@ -22,28 +23,37 @@
...
         return modelMapper.map(groupDto, Group.class);
     }
 
+
     private GroupDto convertGroupToGroupDto(Group group) {
+
```
Comment (MichelRogov) [inline src/main/java/com/school/project/controller/GroupController.java]: get rid of this line here

File: src/main/java/com/school/project/controller/GroupController.java [hunk 7]
```diff
@@ -22,28 +23,37 @@
...
+    }
+
+    @PutMapping("/group/{groupId}/add/{userId}")
+    public void addUserToGroup(@PathVariable Long groupId, @PathVariable Long userId) throws NotFoundException {
+        ResponseEntity.ok().body(convertGroupToGroupDto(groupService.addUser(groupId, userId)));
     }
 
     @GetMapping("/group/{id}")
-    public ResponseEntity<GroupDto> getGroupByID(@PathVariable long id) {
+    public ResponseEntity<GroupDto> getGroupById(@PathVariable Long id) throws NotFoundException {
```
Comment (MichelRogov) [inline src/main/java/com/school/project/controller/GroupController.java]: rest endpoint throws an exception? why? who is catching it? and how?

File: src/main/java/com/school/project/controller/GroupController.java [hunk 8]
```diff
@@ -22,28 +23,37 @@
...
     @PostMapping("/group")
-    public Group createGroup(@RequestBody GroupDto groupDto) {
-        return groupService.create(convertGroupDtoToGroup(groupDto));
+    public ResponseEntity<GroupDto> createGroup(@RequestBody GroupDto groupDto) {
+        return ResponseEntity.ok()
+                .body(convertGroupToGroupDto(groupService.createUpdate(convertGroupDtoToGroup(groupDto))));
+    }
+
+    @PutMapping("/group/{groupId}/add/{userId}")
+    public void addUserToGroup(@PathVariable Long groupId, @PathVariable Long userId) throws NotFoundException {
```
Comment (MichelRogov) [inline src/main/java/com/school/project/controller/GroupController.java]: exception?

File: src/main/java/com/school/project/service/impl/GroupServiceImpl.java [hunk 9]
```diff
@@ -16,23 +18,45 @@
     @Autowired
     GroupRepository groupRepository;
 
+    @Autowired
+    UserRepository userRepository;
+
     @Override
-    public Group create(Group group) {
+    public Group createUpdate(Group group) {
+
```
Comment (MichelRogov) [inline src/main/java/com/school/project/service/impl/GroupServiceImpl.java]: delete this line

File: src/main/java/com/school/project/service/impl/GroupServiceImpl.java [hunk 10]
```diff
@@ -16,23 +18,45 @@
...
         return groupRepository.save(group);
     }
 
     @Override
-    public Group getGroupById(Long id) {
+    public Group getById(Long id) throws NotFoundException {
+
```
Comment (MichelRogov) [inline src/main/java/com/school/project/service/impl/GroupServiceImpl.java]: delete this line

File: src/main/java/com/school/project/service/impl/GroupServiceImpl.java [hunk 11]
```diff
@@ -16,23 +18,45 @@
...
 
     @Override
-    public List<Group> getAllGroups() {
+    public List<Group> getAll() {
         return groupRepository.findAll();
     }
 
+    @Override
+    public Group addUser(Long groupId, Long userId) throws NotFoundException {
+
```
Comment (MichelRogov) [inline src/main/java/com/school/project/service/impl/GroupServiceImpl.java]: delete this line

File: src/main/java/com/school/project/service/impl/GroupServiceImpl.java [hunk 12]
```diff
@@ -16,23 +18,45 @@
...
     }
 
     @Override
-    public List<Group> getAllGroups() {
+    public List<Group> getAll() {
         return groupRepository.findAll();
     }
 
+    @Override
+    public Group addUser(Long groupId, Long userId) throws NotFoundException {
```
Comment (MichelRogov) [inline src/main/java/com/school/project/service/impl/GroupServiceImpl.java]: who is catching this exception? why do you need to throw it?

File: src/main/java/com/school/project/service/impl/GroupServiceImpl.java [hunk 13]
```diff
@@ -16,23 +18,45 @@
...
     @Override
-    public Group create(Group group) {
+    public Group createUpdate(Group group) {
+
         return groupRepository.save(group);
     }
 
     @Override
-    public Group getGroupById(Long id) {
+    public Group getById(Long id) throws NotFoundException {
```
Comment (MichelRogov) [inline src/main/java/com/school/project/service/impl/GroupServiceImpl.java]: who is catching this exception? why do you need to throw it?

File: src/main/java/com/school/project/service/impl/GroupServiceImpl.java [hunk 14]
```diff
@@ -16,23 +18,45 @@
...
+        Optional<User> userById = userRepository.findById(userId);
+        validationFinderById(userById, "User", userId);
+
+        Group group = getById(groupId);
+        List<User> users = group.getUserList();
+        users.add(userById.get());
+
+        return groupRepository.save(group);
+    }
+
```
Comment (MichelRogov) [inline src/main/java/com/school/project/service/impl/GroupServiceImpl.java]: delete this line

File: src/main/java/com/school/project/service/impl/GroupServiceImpl.java [hunk 15]
```diff
@@ -16,23 +18,45 @@
...
+
+    public void validationFinderById(Optional optional, String name, Long id) throws NotFoundException {
+        if (!optional.isPresent())
+            throw new NotFoundException(name + "with id : " + id + " is not found");
+    }
+
```
Comment (MichelRogov) [inline src/main/java/com/school/project/service/impl/GroupServiceImpl.java]: delete this line

File: src/main/java/com/school/project/service/impl/GroupServiceImpl.java [hunk 16]
```diff
@@ -16,23 +18,45 @@
...
+        Group group = getById(groupId);
+        List<User> users = group.getUserList();
+        users.add(userById.get());
+
+        return groupRepository.save(group);
+    }
+
+
+    public void validationFinderById(Optional optional, String name, Long id) throws NotFoundException {
```
Comment (MichelRogov) [inline src/main/java/com/school/project/service/impl/GroupServiceImpl.java]: no need for this function

File: src/main/java/com/school/project/service/impl/GroupServiceImpl.java [hunk 17]
```diff
@@ -16,23 +18,45 @@
...
-    public List<Group> getAllGroups() {
+    public List<Group> getAll() {
         return groupRepository.findAll();
     }
 
+    @Override
+    public Group addUser(Long groupId, Long userId) throws NotFoundException {
+
+        Optional<User> userById = userRepository.findById(userId);
+        validationFinderById(userById, "User", userId);
```
Comment (MichelRogov) [inline src/main/java/com/school/project/service/impl/GroupServiceImpl.java]: why not to reuse user service? you have this logic already

File: src/main/java/com/school/project/service/impl/GroupServiceImpl.java [hunk 18]
```diff
@@ -16,23 +18,45 @@
...
 
     @Override
-    public Group getGroupById(Long id) {
+    public Group getById(Long id) throws NotFoundException {
+
         Optional<Group> byId = groupRepository.findById(id);
-        if (!byId.isPresent()) {
-            throw new GroupNotFoundException("Group with id " + id + " was not found");
-        }
+        validationFinderById(byId, "Group", id);
```
Comment (MichelRogov) [inline src/main/java/com/school/project/service/impl/GroupServiceImpl.java]: use group own not found exception for now

File: src/main/java/com/school/project/controller/GroupController.java [hunk 19]
```diff
@@ -43,6 +50,7 @@ private Group convertGroupDtoToGroup(GroupDto groupDto) {
         return modelMapper.map(groupDto, Group.class);
     }
 
+
```
Comment (MichelRogov) [inline src/main/java/com/school/project/controller/GroupController.java]: why are you adding all this new lines across the project? pls double check all your changes and delete all double lines everywhere except imports (this zone is handled by IDE)

File: src/main/java/com/school/project/dto/GroupDto.java [hunk 20]
```diff
@@ -14,9 +13,9 @@
...
 
-    private Date startDate;
+    private Long startDate;
 
     private Long moduleId;
 
-    private List<UserDto> users;
-}
\ No newline at end of file
+    private List<UserDto> userList;
```
Comment (MichelRogov) [inline src/main/java/com/school/project/dto/GroupDto.java]: after last line in class should be new line

File: src/main/java/com/school/project/service/impl/GroupServiceImpl.java [hunk 21]
```diff
@@ -16,13 +19,19 @@
     @Autowired
     GroupRepository groupRepository;
 
+    @Autowired
+    UserRepository userRepository;
```
Comment (MichelRogov) [inline src/main/java/com/school/project/service/impl/GroupServiceImpl.java]: do you use it somewhere? can't find any usage

File: src/test/java/com/school/project/controller/GroupControllerTest.java [hunk 22]
```diff
@@ -38,37 +42,99 @@
...
+        DateFormat format = new SimpleDateFormat("yyyy-MM-dd");
+        date = format.parse("2019-05-04");
+        Group group = new Group(null, date,
+                null);
+        group.setId(2L);
+
+        return group;
+    }
+
+
```
Comment (MichelRogov) [inline src/test/java/com/school/project/controller/GroupControllerTest.java]: double line

File: src/main/java/com/school/project/service/impl/GroupServiceImpl.java (hunk 21 above)
Comment (LeSitar) [reply src/main/java/com/school/project/service/impl/GroupServiceImpl.java]: yes, i use it here:
 @Override
    public Group addUser(Long groupId, Long userId) {
        User userById = userService.getUserById(userId);  (line 49)

**Pull request merged**